### ISSUE TITLE
Array knob of empty string converts to an array of [""]

### ISSUE THREAD

**Issue by pmwisdom** [storybooks/storybook#1404]
When I add an empty array knob to a component, aka:

`array('Label', [])` it shows correctly in the knob panel that its empty, and the component receives the prop `[]`, but when I add a comma delimited string to the knob input (say, `hello, world`, which converts to `['hello', 'world']` correctly) and then delete the string from the knob panel (back to `""`) my component receives `[""]`, when it should receive `[]`.

My solution for this is to create a patched component, but I believe this to be a bug.

Example workaround:

```` javascript
//patched-array.js
import {array} from '@storybook/addon-knobs';

const removeEmptyEntries = arr => {
	return arr.filter(e => Boolean(e));
};

const patchedArray = (name, value) => {
	return removeEmptyEntries(array(name, value));
};

export default patchedArray;
````

**Comment by dangreenisrael** [comment]
Should this not be closed?

**Comment by Hypnosphi** [comment]
No, it's not yet released as stable

**Comment by dangreenisrael** [comment]
Makes sense, thanks @Hypnosphi. Thanks for the clarification

**Comment by Hypnosphi** [comment]
Fixed in `3.3.0-alpha`
